ui/src/components/credentials/IssueCredentialForm.tsx

```diff
@@ -213,182 +215,266 @@ export function IssueCredentialForm({
+            const parsedDefault = z.number().safeParse(curr.schema.default);
+            const constValue = parsedConst.success ? parsedConst.data : undefined;
+            const defaultValue = parsedDefault.success ? parsedDefault.data : undefined;
+            const value = constValue !== undefined ? constValue : defaultValue;
+            return { ...acc, [curr.name]: value };
+          }
+          case "string": {
+            const parsedConst = z.string().safeParse(curr.schema.const);
+            const parsedDefault = z.string().safeParse(curr.schema.default);
+            const constValue = parsedConst.success ? parsedConst.data : undefined;
+            const defaultValue = parsedDefault.success ? parsedDefault.data : undefined;
+            const value = constValue !== undefined ? constValue : defaultValue;
+            if (value === undefined) {
+              return acc;
+            }
+            switch (curr.schema.format) {
+              case "date":
+              case "date-time": {
+                return { ...acc, [curr.name]: dayjs(value) };
+              }
+              case "time": {
+                return { ...acc, [curr.name]: dayjs(`1978-01-29T${value}`) };
+              }
+              default: {
+                return { ...acc, [curr.name]: value };
+              }
+            }
+          }
+          case "object": {
+            const parsedRecord = z.record(z.unknown()).safeParse(initialValues[curr.name] || {});
+            return parsedRecord.success
+              ? {
+                  ...acc,
+                  [curr.name]: computeFormObjectInitialValues(curr, parsedRecord.data),
+                }
+              : acc;
+          }
+          default: {
+            return acc;
+          }
+        }
+      },
+      initialValues
+    );
+  };
+
   useEffect(() => {
     const { aborter } = makeRequestAbortable(fetchSchemas);
 
     return aborter;
   }, [fetchSchemas]);
 
+  const credentialSubjectAttributeWithoutId = isAsyncTaskDataAvailable(jsonSchema)
+    ? extractCredentialSubjectAttributeWithoutId(jsonSchema.data)
+    : undefined;
+
+  const initialValuesWithSchemaValues = credentialSubjectAttributeWithoutId && {
+    ...initialValues,
+    credentialSubject: computeFormObjectInitialValues(
+      credentialSubjectAttributeWithoutId,
+      initialValues.credentialSubject || {}
+    ),
+  };
+
   return (
     <>
       {messageContext}
 
-      <Form
-        initialValues={initialValues}
-        layout="vertical"
-        onFinish={(values: IssueCredentialFormData) => {
-          const jsonSchemaData = isAsyncTaskDataAvailable(jsonSchema) ? jsonSchema.data : undefined;
-          const credentialSubjectAttributeWithoutId =
-            jsonSchemaData && extractCredentialSubjectAttributeWithoutId(jsonSchemaData);
-
-          if (
-            jsonSchemaData &&
-            credentialSubjectAttributeWithoutId &&
+      {initialValuesWithSchemaValues ? (
```

Comment:
I see one problem here. If the `credentialSubjectAttributeWithoutId` is `undefined` because the `jsonSchema` failed to load, it will make this `initialValuesWithSchemaValues` to be `false`, so the `<LoadingResult />` will be shown forever.

I think that instead, this `initialValuesWithSchemaValues` variable should be an `AsyncTask` to handle the errors properly in the view.

Reply:
Very good catch! This is indeed an ugly bug that breaks the app when there's no schema id in the URL, since the schema selector is not shown when this is the case.

Reply:
Fixed. Loader removed. Now we set the values to the form with a `useEffect` when they are available.

Thx for catching this one! 😅